router/router.go

```diff
@@ -1054,14 +1054,14 @@ func (worker *workerT) postStatusOnResponseQ(respStatusCode int, respBody string
 		if worker.rt.guaranteeUserEventOrder {
 			if addToFailedMap {
 				//#JobOrder (see other #JobOrder comment)
-				worker.failedJobIDMutex.RLock()
+				worker.failedJobIDMutex.Lock()
```

Comment:
Any thoughts on this? 

```
			worker.failedJobIDMutex.Lock()
			if addToFailedMap {
				//#JobOrder (see other #JobOrder comment)
                //Do something
            } else {
                // Do something
             }
			worker.failedJobIDMutex.Unlock()
```

Reply:
Are you suggesting that we should check `addToFailedMap` within the mutex lock? Isn't that beyond the scope of that mutex?

Reply:
Yeah it is outside the scope but I was just wondering if we can avoid doing `worker.failedJobIDMutex.Unlock()` twice